lib/mirage/git_mirage.ml

```diff
@@ -118,20 +114,19 @@ module FS (FS: FS) (D: Git.Hash.DIGEST) (I: Git.Inflate.S) = struct
 
     let read_file t file =
       Log.debug (fun l -> l "read_file %s" file);
-      FS.stat t file >>| fun s ->
+      FS.stat t file >>= get_fs_error >>= fun s ->
       is_directory t file >>= function
       | false ->
-        FS.read t file 0 (Int64.to_int s.FS.size) >>| fun bs ->
+        FS.read t file 0 (Int64.to_int s.FS.size) >>= get_fs_error >|= fun bs ->
```

Comment:
I don't know much, but this feels slightly wrong ([read](https://github.com/mirage/mirage/blob/f09a53eacd53c28438c78a5ce647c550274f546e/types/V1.mli#L909) should take two `int64`).

Also, `Int64.to_int` is pretty much unsafe.